### PR TITLE
feat: toolbar update and css layout

### PR DESCRIPTION
**Pull request by jenniferarnesen**
Changes in the following files are functional changes needed for the new Analytics toolbar:
* src/components/app/AppMenu.js - add the `HoverMenuBar` component and onFileMenuAction callback, which updates the interpretations panel with changes (e.g. sharing, renaming, translating) if it is open
* AppLayout - `onFileMenuAction` and the `interpretationRenderId`.
* FileMenu - added `onFileMenuAction`
* Interpretations, InterpretationsPanel:  - added `renderId` which triggers refetching the map info if it was changed by a file menu action
* InterpretationsToggle:  use the new `InterpretationsAndDetailsToggler` from Analytics
* DownloadButton: needed different style to look nice with the new toolbar
* AddLayerButton: styled to fit into the new toolbar

The new toolbar is a different height than the old one, which caused a bunch of layout issues due to the absolute positioning. Switched to `flex` where it made sense, and css changes accordingly.

Only change to DownloadSettings is the addition of a wrapping div `<div className={styles.downloadSettingsPanel}>`
MapPosition: resize due to open/close of the Layers or Details panel has a timeout added, since there is now a 100ms transition when open/close those Drawers

### CODE REVIEW

**Review by turban — COMMENTED** [review]
Some issues with the download mode:

1. Reduce margin below the map
2. The map controls (top right) should not show in download mode

Screenshot 2023-09-09 at 13 17 24

File: src/components/app/App.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ import { CURRENT_AO_KEY } from '../../util/analyticalObject.js'
 import { getUrlParameter } from '../../util/requests.js'
 import { useSystemSettings } from '../SystemSettingsProvider.js'
 import AppLayout from './AppLayout.js'
-import styles from './styles/App.module.css'
+import './App.css'
```
Comment (turban) [inline src/components/app/App.js:15]: Possible to add these varaibles to App.module.css to keep them in one place?

File: src/components/app/AppLayout.js [hunk 2]
```diff
@@ -1,49 +1,64 @@
...
 import MapPosition from '../map/MapPosition.js'
 import OpenAsMapDialog from '../openAs/OpenAsMapDialog.js'
-import OrgUnitProfile from '../orgunits/OrgUnitProfile.js'
 import AppMenu from './AppMenu.js'
+import DetailsPanel from './DetailsPanel.js'
 import styles from './styles/AppLayout.module.css'
 
 const AppLayout = () => {
-    const downloadMode = useSelector((state) => state.download.downloadMode)
+    const [interpretationsRenderId, setInterpretationsRenderId] = useState(1)
```
Comment (turban) [inline src/components/app/AppLayout.js:19]: Rename to `interpretationsRenderCount`?

File: src/components/app/AppLayout.js [hunk 3]
```diff
@@ -1,49 +1,64 @@
...
+
+    const dataTableOpen = useSelector((state) => !!state.dataTable)
+    const downloadModeOpen = useSelector(
+        (state) => !!state.download.downloadMode
```
Comment (turban) [inline src/components/app/AppLayout.js:23]: "downloadMode" is a boolean so `!!` is not needed

File: src/components/app/AppLayout.js [hunk 4]
```diff
@@ -1,49 +1,64 @@
...
+    )
+    const detailsPanelOpen = useSelector(
+        (state) => state.ui.rightPanelOpen && !state.orgUnitProfile
+    )
+
+    const onFileMenuAction = () =>
+        detailsPanelOpen &&
+        setInterpretationsRenderId(interpretationsRenderId + 1)
```
Comment (turban) [inline src/components/app/AppLayout.js:31]: Change to `setInterpretationsRenderCount`?

File: src/components/app/DetailsPanel.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    const getContent = () => {
+        if (!detailsPanelOpen) {
+            return null
+        }
+
+        return viewOrgUnitProfile ? (
+            <OrgUnitProfile />
+        ) : (
+            <Interpretations renderId={interpretationsRenderId} />
```
Comment (turban) [inline src/components/app/DetailsPanel.js:21]: Change to `renderCount`?

**Comment by turban** [comment]
Not sure if this is a bug or not - if I open a map, and save it - it says last updated 2 hours ago. 

Screenshot 2023-09-11 at 14 51 27

Maybe "2 hours" is the most recent time?

File: src/components/app/App.js (hunk 1 above)
Comment (jenniferarnesen) [reply src/components/app/App.js:15]: I think that global variables like the ones in App.css have to be defined in a regular css file, not a css module, which is specific to a class and doesn't have global scope.

**Comment by jenniferarnesen** [comment]
> Not sure if this is a bug or not - if I open a map, and save it - it says last updated 2 hours ago.
> 
> Screenshot 2023-09-11 at 14 51 27
> Maybe "2 hours" is the most recent time?

That bug is solved in https://github.com/dhis2/analytics/pull/1574, which is why we have to raise the priority on the toolbar update (analytics 26)

File: src/components/interpretations/InterpretationsPanel.js [hunk 6]
```diff
@@ -36,10 +41,11 @@ const InterpretationsPanel = ({ interpretationId, map, setInterpretation }) => {
         setInterpretation()
     }, [setInterpretation])
 
+    console.log('renderCount', renderCount)
```
Comment (turban) [inline src/components/interpretations/InterpretationsPanel.js:44]: Remove console.log

**Review by turban — APPROVED** [review]

**Pull request merged**